plenum/test/restart/test_restart_to_lower_view.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+
+from plenum.test import waits
+from plenum.test.helper import sdk_send_random_and_check
+from plenum.test.node_request.helper import sdk_ensure_pool_functional
+from plenum.test.restart.helper import restart_nodes
+from plenum.test.view_change.helper import ensure_view_change_complete
+
+nodeCount = 4
+
+
+@pytest.fixture(scope="module")
+def tconf(tconf):
+    old_network_3pc_watcher_state = tconf.ENABLE_NETWORK_I3PC_WATCHER
+    tconf.ENABLE_NETWORK_I3PC_WATCHER = True
+    yield tconf
+    tconf.ENABLE_NETWORK_I3PC_WATCHER = old_network_3pc_watcher_state
+
+
+def test_restart_majority_to_lower_view(looper, txnPoolNodeSet, tconf, tdir, allPluginsPath,
+                                        sdk_pool_handle, sdk_wallet_client):
+    # Add transaction to ledger
+    sdk_send_random_and_check(looper, txnPoolNodeSet, sdk_pool_handle, sdk_wallet_client, 1)
+
+    # Move to higher view
+    ensure_view_change_complete(looper, txnPoolNodeSet)
+
+    majority = txnPoolNodeSet[:3]
+    minority = txnPoolNodeSet[3:]
+
+    # Restart majority group
+    tm = tconf.ToleratePrimaryDisconnection + waits.expectedPoolElectionTimeout(len(txnPoolNodeSet))
+    restart_nodes(looper, txnPoolNodeSet, majority, tconf, tdir, allPluginsPath,
```

Comment:
Because they don't finish (or at least `ensure_elections_done` is not satisfied with result) and test just fails before it has chance to check that inconsistent 3pc state was indeed detected.

Reply:
Probably `ensureElectionsDone` looking only at majority of nodes will still pass, although I'm not sure whether we should call it here manually or improve `restart_nodes` or even `ensureElectionsDone` itself.

Reply:
Done